src/webapi/history.rs

```diff
@@ -0,0 +1,64 @@
+#![cfg(feature = "serde")]
+use webcore::value::Reference;
+use webcore::try_from::TryInto;
+use webcore::serialization::JsSerializable;
+
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/History)
+pub struct History(Reference);
+
+reference_boilerplate! {
+    History,
+    instanceof History
+}
+
+impl History {
+    /// `History.pushState` allows you to add history entries.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/History_API#The_pushState%28%29_method)
+    pub fn push_state<T: JsSerializable>(&self, state: T, title: Option<&str>, url: String) {
```

Comment:
In the [specs](https://html.spec.whatwg.org/multipage/history.html#the-history-interface) I can see that the `title` is marked as being non-optional (is not inside square brackets) while the `url` is, so shouldn't the prototype be `&self, state: T, title: &str, url: Option< &str >`?